### DISCUSSION TITLE
Add `eslint-config-unifiedjs` with `eslint-config-xo` and other plugins

### DISCUSSION THREAD

**Discussion by JounQin** [unifiedjs/unified#198] [Ideas]
`eslint-config-xo` only checks `.js` and `.ts` files, with other plugins, we can check `.json`, `.yml`, etc. and more!

**Comment by remcohaszing** [comment]
Before we make any decisions, I think we should wait to see how the new ESLint config system works out, and how it will relate to XO.

- [ESLint's new config system, Part 1: Background](https://eslint.org/blog/2022/08/new-config-system-part-1/)
- [ESLint's new config system, Part 2: Introduction to flat config](https://eslint.org/blog/2022/08/new-config-system-part-2/)
- [ESLint's new config system, Part 3: Developer preview](https://eslint.org/blog/2022/08/new-config-system-part-3/)

While I personally prefer a linting setup more strict than XO, and ESLint related tooling is also more mature, I have my doubts about using it for anything other than JavaScript (including TypeScript).

**Comment by wooorm** [comment]
> we can check .json, .yml, etc. and more!

Two questions:

a) What would we use to check that? Which plugins and rules are possible
b) Are there examples in unified of YAML, JSON, etc files, that you’d like to check, which currently are chaotic / have stylistic problems / have bugs?

**Reply by remcohaszing** [reply]
I personally like [`prettier-plugin-packagejson`](https://github.com/matzkoh/prettier-plugin-packagejson) for sorting `package.json`. No fuzz, just install it and Prettier will use it. I personally find it a bit overkill for non-mono-repositories, but since unified tries to maintain a consistent sorting order in their packages, I think it could add value.

**Reply by JounQin** [reply]
> a) What would we use to check that? Which plugins and rules are possible

For example, we can check `package.json` and GitHub workflows `.yml` files with https://github.com/ota-meshi/eslint-plugin-json-schema-validator. Custom schema with `$schema` is also supported.

> b) Are there examples in unified of YAML, JSON, etc files, that you’d like to check, which currently are chaotic / have stylistic problems / have bugs?

There are a lot of `prettier` plugins for `package.json` like https://github.com/matzkoh/prettier-plugin-packagejson, https://github.com/shellscape/prettier-plugin-package or https://github.com/un-ts/prettier/tree/master/packages/pkg.

**Reply by wooorm** [reply]
The last point didn’t really answer what I tried to ask. Trying to rephrase: do you know of JSON or YAML files in our repos that are not consistent?

I am asking because, yes, consistency is important. But I believe the files are already consistent.

Does it make some sense for some projects? Sure.
Does it add much to the 300 projects we maintain? I am not sure.

**Reply by JounQin** [reply]
> I am asking because, yes, consistency is important. But I believe the files are already consistent.

It depends on a human writing, which is possible to be inconsistent, I suppose, although I didn't try to find out because it could be an impossible job to check all repositories.

> Does it make some sense for some projects? Sure.
> Does it add much to the 300 projects we maintain? I am not sure.

We have tools to ensure consistency, we can migrate one by one, not at a time, If we agree to migrate of course. Contributors are welcome to help.

**Reply by wooorm** [reply]
> although I didn't try to find out

This is very important to me.
You want to change something.
You want to add a tool that sorts `package.json`s.
The goal is to make files more consistent.
It is unclear if this tool makes files more consistent.

Like any another proposal, I think for this proposal to be accepted, it should be clear that it improves actual things.

> it could be an impossible job to check all repositories.

Now imagine how long it takes to make changes to all projects 😉
You don’t have to check everything.
You can check some? Say, 4 or 10? See if it improves things?

> We have tools to ensure consistency,

We could also add a tool that sorts `.gitignore` files.
But does it improve the situation?
In most cases, they are already consistent.
In some other cases, they are sorted a bit weird.
But does it matter?

> Contributors are welcome to help.

Yeah, and that would help a lot: for someone to take this on. But it’s a lot of work. I am not sure if someone is interested in updating 100s of repos.
